lib/rondo_form/view_helpers.rb

```diff
@@ -54,17 +57,19 @@ def link_to_add_association(*args, &block)
         new_object = f.object.class.reflect_on_association(association).klass.new
         model_name = new_object.class.name.underscore
         hidden_div = content_tag("template", id: "#{model_name}_fields_template", data: {'nested-rondo_target': 'template'}) do
-          render_association(association, f, new_object, html_options)
+          render_association(association, f, new_object, render_options)
         end
         hidden_div.html_safe + link_to(name, '', html_options )
       end
     end
 
     # :nodoc:
-    def render_association(association, f, new_object, html_options)
-      partial_name = html_options[:partial_name] || association.to_s.singularize + "_fields"
+    def render_association(association, f, new_object, render_options)
+      locals = render_options.delete(:locals) || {}
+      render_options[:partial] = "#{association.to_s.singularize}_fields" unless render_options[:partial]
       f.fields_for(association, new_object, :child_index => "new_#{association}") do |builder|
-        render(partial_name, :f => builder, :dynamic => true)
```

Comment:
I dropped `:dynamic => true` because I couldn't find any reference to it in the Rails docs or source, but if it's a local variable in your partials you can just pass it with the locals.